calc/l3_route_resolver.go

```diff
@@ -688,17 +709,54 @@ func (r *RouteTrie) RemoveHost(cidr ip.V4CIDR, nodeName string) {
 	})
 }
 
-func (r *RouteTrie) AddLocalWEP(cidr ip.V4CIDR) {
+func (r *RouteTrie) AddWEP(cidr ip.V4CIDR, nodename string) {
 	r.updateCIDR(cidr, func(ri *RouteInfo) {
-		ri.LocalWEP.RefCount++
+		if ri.WEP.RefCount == nil {
+			ri.WEP.RefCount = map[string]int{}
+		}
+		ri.WEP.RefCount[nodename]++
+
+		// Groom the nodename slice.
```

Comment:
I would like to switch to using a list to store the WEPs, like you suggested. This was quicker and gets the same result IIUC, and I'd like to progress with testing, so I went with this for now.

Should be relatively seamless to switch the implementation to a `[]struct{}` rather than a map once I've got some more tests written.